lib/spack/spack/environment/environment.py

```diff
@@ -259,7 +259,8 @@ def active_environment() -> Optional["Environment"]:
 
 def _root(name):
     """Non-validating version of root(), to be used internally."""
-    return os.path.join(env_root_path(), name)
+    path = os.path.join(env_root_path(), name)
+    return os.readlink(path) if islink(path) else path
```

Comment:
One FS op instead of two:

```suggestion
    try:
        return os.readlink(path)
    except OSError:
        return path
```